synapse/handlers/sync.py

```diff
@@ -770,7 +770,7 @@ def compute_state_delta(self, room_id, batch, sync_config, since_token, now_toke
                     current=current_state_ids,
                     lazy_load_members=lazy_load_members,
                 )
-            elif batch.limited:
+            elif batch.limited and batch.events:
```

Comment:
To me it sounds like the error in https://github.com/matrix-org/synapse/issues/4842 was due to `batch.events` being an empty list, rather than `None`. In that case, this `if` statement would still run and crash.

Instead you may want to check if `len(batch.events) > 0`.

Reply:
I'm not a Python expert. I found <https://stackoverflow.com/questions/53513/how-do-i-check-if-a-list-is-empty>